ignition-server/controllers/local_ignitionprovider.go

```diff
@@ -403,6 +403,63 @@ func (p *LocalIgnitionProvider) GetPayload(ctx context.Context, releaseImage, cu
 		return nil, fmt.Errorf("failed to download binaries: %w", err)
 	}
 
+	// As of OpenShift 4.20, the cluster-authentication-operator is responsible for managing
+	// the RoleBindingRestriction CRD and has been removed from the cluster-config-api
+	// payload image.
+	// Because this CRD is required for bootstrapping, ensure we use the cluster-authentication-operator
+	// payload image to get the RoleBindingRestriction CRD in the set of bootstrap manifests.
+	if payloadVersion.Major == 4 && payloadVersion.Minor >= 20 {
+		// set the component to the correct binary name and file path based on the payload version
+		clusterAuthenticationOperatorComponent := "cluster-authentication-operator"
+		clusterAuthenticationOperatorManifestRendererFile := "usr/bin/authentication-operator"
+		err = func() error {
+			start := time.Now()
+			clusterAuthenticationOperatorImage, ok := imageProvider.ImageExist(clusterAuthenticationOperatorComponent)
+			if !ok {
+				return fmt.Errorf("release image does not contain $%s (images: %v)", clusterAuthenticationOperatorComponent, imageProvider.ComponentImages())
+			}
+
+			clusterAuthenticationOperatorImage, err = registryclient.GetCorrectArchImage(ctx, clusterAuthenticationOperatorComponent, clusterAuthenticationOperatorImage, pullSecret, p.ImageMetadataProvider)
+			if err != nil {
+				return err
+			}
+
+			log.Info(fmt.Sprintf("discovered image %s image %v", clusterAuthenticationOperatorComponent, clusterAuthenticationOperatorImage))
+
+			// Making sure image uses the registry override for disconnected environments
+			checkedClusterAuthenticationOperatorImage, err := p.ImageMetadataProvider.GetOverride(ctx, clusterAuthenticationOperatorImage, pullSecret)
+			if err != nil {
+				return err
+			}
+
+			caoComposedImage := checkedClusterAuthenticationOperatorImage.String()
+			if caoComposedImage != clusterAuthenticationOperatorImage {
+				clusterAuthenticationOperatorImage = caoComposedImage
+				log.Info(fmt.Sprintf("using mirrored %s image %v", clusterAuthenticationOperatorComponent, caoComposedImage))
+			}
+
+			file, err := os.Create(filepath.Join(binDir, clusterAuthenticationOperatorComponent))
+			if err != nil {
+				return fmt.Errorf("failed to create file: %w", err)
+			}
+			if err := file.Chmod(0777); err != nil {
```

Comment:
security nit: should we really use 777 here?

Reply:
This logic was mostly copied from directly above for the standard o/api manifest renderer which is `0777` as well.

The only tweaks I had made was image and binary location and variable names (i didn't want to override something in case it was used later).

I contemplated making this a common function to use as well. If this is something folks think would be useful I'm happy to do this as well.

I'm open to making changes here, and if we do I'll make it across the board.

I'll leave it up to those more familiar with the historical context of why `0777` was originally chosen for the existing logic to decide if this needs to change.